src/components/Email/EmailForm.tsx

```diff
@@ -164,19 +190,24 @@ function EmailForm({ setIsEditMode, queryRef, onClose }: Props) {
           autoFocus
           disabled={savePending}
         />
-        <HStack justify="flex-end" align="flex-end" gap={8}>
-          {showCancelButton && (
-            <Button variant="secondary" disabled={savePending} onClick={handleCancelClick}>
-              Cancel
+
+        <HStack align="center" justify={showErrorMessage ? 'space-between' : 'flex-end'}>
+          {showErrorMessage && <ErrorText message={"The email doesn't appear to be valid"} />}
+
+          <HStack gap={8}>
+            {showCancelButton && (
+              <Button variant="secondary" disabled={savePending} onClick={handleCancelClick}>
+                Cancel
+              </Button>
+            )}
+            <Button
+              variant="primary"
+              disabled={!isValidEmail || savePending || savedEmail === email || isChecking}
+              onClick={handleSaveClick}
+            >
+              {isChecking ? <Spinner /> : 'Save'}
```

Comment:
could we make the button width constant when the spinner appears? just to prevent layout shift!

in fact, i think we can use the built-in `pending` prop for buttons!

<img width="435" alt="Screen Shot 2022-12-23 at 3 01 27 PM" src="https://user-images.githubusercontent.com/12162433/209400993-edb581cf-88c3-4fd3-82c1-69207d5b3ead.png">

Reply:
Thanks @Robinnnnn! Somehow I missed this props 🤦